src/store/Controller.cc

```diff
@@ -855,49 +858,51 @@ Store::Controller::syncCollapsed(const sfileno xitIndex)
     collapsed->setCollapsingRequirement(true);
 }
 
-/// Called for Transients entries that are not yet anchored to a cache.
+/// If possible and has not been done, associates the entry with its store(s).
 /// \returns false for not-yet-cached entries that we may attach later
 /// \returns true for other entries after synchronizing them with their store
-/// and setting inSync to reflect that synchronization outcome.
 bool
-Store::Controller::anchorToCache(StoreEntry &entry, bool &inSync)
+Store::Controller::anchorToCache(StoreEntry &entry)
 {
     assert(entry.hasTransients());
     assert(transientsReader(entry));
 
+    // TODO: Attach entries to both memory and disk
+
+    // TODO: Reduce code duplication with syncCollapsed()
```

Comment:
This TODO can probably be addressed without a lot of changes leaking outside Controller.cc, but doing so will require refactoring syncCollapsed() quite a bit, including how it (and its callers) handle exceptions. That refactoring is best done in a dedicated PR or, at least, when we have to fix some syncCollapsed() bug. This PR fixes an anchorCollapsed() bug. It is best to keep this PR as focused on that fix/bug as possible.